src/main/java/org/prebid/server/bidder/pubnative/PubnativeBidder.java

```diff
@@ -98,17 +118,22 @@ private static BidRequest modifyRequest(BidRequest bidRequest, Imp imp) {
                 }
 
                 final Format firstFormat = bannerFormats.get(0);
-                final Banner modifiedBanner = banner.toBuilder()
+                return banner.toBuilder()
                         .h(firstFormat.getH())
                         .w(firstFormat.getW())
                         .build();
-                outgoingImp = imp.toBuilder().banner(modifiedBanner).build();
             }
         }
+        return banner;
+    }
 
-        return bidRequest.toBuilder()
-                .imp(Collections.singletonList(outgoingImp))
-                .build();
+    private BigDecimal resolveBidFloor(BidRequest bidRequest, Imp imp) {
+        final String bidFloorCur = imp.getBidfloorcur();
+        final BigDecimal bidFloor = imp.getBidfloor();
+        if (bidFloor == null || bidFloor.compareTo(BigDecimal.ZERO) <= 0 || StringUtils.isEmpty(bidFloorCur)) {
```

Comment:
i'd say `bidFloorCur` is not required here to check. It can be assumed from `request.cur`.

@snahornyi pls confirm.

Reply:
Yes, request.cur[0] will be assumed as adServerCurrency